src/components/duelling-picklist/duelling-picklist.test.js

```diff
@@ -0,0 +1,990 @@
+        removeButton(0).focus().trigger("keydown", keyCode(pressed));
+        unassignedPicklistItems().should("have.length", 10);
+        assignedPicklistItems().should("have.length", 0);
+      }
+    );
+
+    it.each([
+      ["Content", 10],
+      ["Content 1", 2],
+      ["Content 10", 1],
+    ])(
+      "should verify when %s is enterted into search field that %s results are displayed",
+      (searchString, results) => {
+        CypressMountWithProviders(<DuellingPicklistComponent />);
+
+        duellingSearchInput().eq(0).type(searchString);
+        unassignedPicklistItems().should("have.length", results);
+      }
+    );
+
+    it("should verify leftControl prop in component generates search field appears above unassigned picklist", () => {
+      CypressMountWithProviders(<DuellingPicklistComponent />);
+
+      getDataElementByValue("picklist-left-control")
+        .children()
+        .should("have.attr", "data-component", "search");
+    });
+
+    it("should verify rightControl prop in component generates search field appears above assigned picklist", () => {
+      CypressMountWithProviders(<DuellingPicklistComponent />);
+
+      getDataElementByValue("picklist-right-label")
+        .children()
+        .should("have.attr", "data-component", "search");
+    });
+
+    it.each([
+      ["disabled", true, "have.attr"],
+      ["enabled", false, "not.have.attr"],
+    ])(
+      "should verify Duelling Picklist is %s when disabled prop is %s",
+      (state, bool, attribute) => {
+        CypressMountWithProviders(
+          <DuellingPicklistComponent disabled={bool} />
+        );
+
+        duellingPicklistComponent().should(attribute, "disabled");
+      }
+    );
+
+    it("should verify unassigned picklist label is 'Left Label'", () => {
+      CypressMountWithProviders(
+        <DuellingPicklistComponent leftLabel="Left Label" />
+      );
+
+      picklistLeftLabel().should("have.text", "Left Label");
+    });
+
+    it("should verify assigned picklist label is 'Right Label'", () => {
+      CypressMountWithProviders(
+        <DuellingPicklistComponent rightLabel="Right Label" />
+      );
+
+      picklistRightLabel().should("have.text", "Right Label");
+    });
+  });
+
+  describe("should render Duelling-Picklist component to test Picklist props", () => {
+    it.each(specialCharacters)(
+      "should verify picklist placeholder = %s",
+      (chars) => {
+        CypressMountWithProviders(
+          <DuellingPicklistComponentPicklistProps
+            placeholder={<PicklistPlaceholder text={chars} />}
+          />
+        );
+
+        for (let i = 0; i < 10; i++) {
+          addButton(0).focus().click();
+        }
+        unassignedPicklist().find("div").should("have.text", chars);
+      }
+    );
+
+    it.each([
+      ["locked", true, "have.attr"],
+      ["unlocked", false, "not.have.attr"],
+    ])(
+      "should verify picklist item is %s when locked prop is %s",
```

Comment:
for `locked` state there is also a colour changing, can you add that assertion as well ?